kfmon.c

```diff
@@ -1481,20 +1479,50 @@ static bool
 					book_path,
 					sizeof(book_path),
 					NOTRUNC);
-				replace_invalid_chars(converted_book_path);
+
+				// Check the database version to decide whether to preserve the dot for the file extension
+				int db_version = 0;
+				CALL_SQLITE(prepare_v2(db, "SELECT version FROM DbVersion LIMIT 1;", -1, &stmt, NULL));
+				rc = sqlite3_step(stmt);
+				if (rc == SQLITE_ROW) {
+					db_version = sqlite3_column_int(stmt, 0);
+					DBGLOG("Database version: %d", db_version);
+				}
+				if (db_version >= 191) {
+					// Implement the 5.6 variant, which preserves the dot for the file extension
+
+					// Separate the extension from the base path
+					char* ext = strrchr(converted_book_path, '.');
+					if (ext) {
+						*ext = '\0'; // Temporarily terminate the string to isolate the base path
+					}
+
+					// Replace invalid characters in the base path
+					replace_invalid_chars(converted_book_path);
+
+					// Reattach the extension if it exists
+					if (ext) {
+						*ext = '.'; // Restore the dot
+					}
+				} else {
+					// Fallback to the v5 variant
+					replace_invalid_chars(converted_book_path);
+				}
+				sqlite3_finalize(stmt);
```

Comment:
As I mentioned, I'm not so good at C. But is sqlite3_finalize correct here? Isn't this done a few lines further down anyway?

Reply:
You are right, I didn't see that there was a sqlite3_finalize statement below anyways. I will remove it here.